luavm/src/lib/instructions/arithmetic_operators.rs

```diff
@@ -0,0 +1,102 @@
+use lua_value::*;
+use luacompiler::bytecode::instructions::{first_arg, second_arg, third_arg};
+use Vm;
+
+/// Perform <op> on the arguments of <instr>. The arguments are converted using
+/// the rules of Lua.
+macro_rules! bin_op {
+    ($vm:ident, $instr: ident, $op: tt) => {
+        {
+            let lhs = &$vm.registers.borrow()[second_arg($instr) as usize];
+            let rhs = &$vm.registers.borrow()[third_arg($instr) as usize];
+            if lhs.is_float() || rhs.is_float() {
+                let float = match (lhs.to_float(), rhs.to_float()) {
+                    (Some(l), Some(r)) => l $op r,
+                    (_, _) => panic!("Failed to convert to float!")
```

Comment:
Does Lua `panic` in such situations? I suspect we need to get some sort of Lua error handling in now, or it'll be very hard to add it in later.

Reply:
It doesn't panic, but gives you an error. I will have to come up with my own `Result` and pass the error up to the vm.

Reply:
OK.